tools/toolchains/gcc.py

```diff
@@ -48,14 +48,19 @@ def __init__(self, target,  notify=None, macros=None, build_profile=None,
 
         tool_path = TOOLCHAIN_PATHS['GCC_ARM']
         # Add flags for current size setting
-        default_lib = "std"
+        default_lib = ""
         if hasattr(target, "default_lib"):
+            self.check_c_lib_supported(target, "gcc_arm")
             default_lib = target.default_lib
         elif hasattr(target, "default_build"):
             default_lib = target.default_build
 
+        if not default_lib:
```

Comment:
`default_lib` does not have to be set to anything. Nothing will happen if it is anything other than "small". You do not need to raise an exception.